### PR TITLE
fix(codegen): loop-carry manual_scope TaskId into Array[TASK_ID]

### LINKED ISSUE #1811: [Bug] codegen: manual_scope TaskId can't be loop-carried into Array[TASK_ID]
### Component

Codegen

### Description

A `TaskId` captured inside `pl.manual_scope()` cannot be loop-carried into a `pl.array(pl.TASK_ID)` across a `pl.range` loop. Orchestration codegen fails with an `INTERNAL_CHECK`:

```
Internal error: scalar yield to array carry must resolve to a TaskId variable registered in manual_task_id_map_
Check failed: tid_binding != nullptr at src/codegen/orchestration/orchestration_codegen.cpp:1389
```

**Root cause.** When a `pl.range` loop carries an `Array[TASK_ID]` whose slots are filled per-iteration by a `pl.parallel` block (the "scalar yield to array carry" path), codegen resolves each yielded tid through `manual_task_id_map_`. A tid produced **inside `pl.manual_scope()`** is not registered there, so the lookup fails. The identical kernel compiles cleanly when the producer runs in the ordinary auto-dep region (no `manual_scope`) — see the contrast below, where the *only* difference is the `with pl.manual_scope():` wrapper.

This looks like the same family as the (closed) #1577 "TaskId array element scalar deps are not registered in auto-scope codegen", but the complementary direction: #1577 was the **read** path (`array.get_element(Array[TASK_ID]) -> Scalar[TASK_ID]` used in `deps=`); this is the **write / loop-carry** path (a `manual_scope` tid yielded into a carried `Array[TASK_ID]`), and it still fails on latest `main` (`3ec5e838`).

**Secondary issue:** `compile_for_test()` *prints* the error but does not raise — the failure only becomes fatal (`PartialCodegenError`) in the device/codegen path. Ideally the frontend would reject this earlier with a clear user-facing diagnostic instead of an `INTERNAL_CHECK`.

**Real-world impact:** in the qwen3-14b decode (`models/qwen3/14b/decode_layer.py`), `dcr_xgamma`'s `DOWN_ON` sliced writers of `out` / `normed_out` WAW-serialize on one core when emitted in the auto-dep region (the `OutWindowExternalizer` Pattern-5 that should keep them parallel bails because the task writes two outputs). Moving the writer **into** `manual_scope` removes the WAW (parallel writers), but is blocked by this codegen limitation because the per-layer dcr tids are loop-carried (`prev_out_tids` / `prev_normed_tids`). We had to fall back to a single `pl.spmd` dispatch instead.

### Steps to Reproduce

Compile-only, no device needed:

```python
import sys
import pypto.language as pl
import torch
from pypto.backend import BackendType, set_backend_type

N, LOOP, ROWS, COLS = 4, 2, 16, 256
TILE = COLS // N

@pl.jit.inline
def layer(
    x: pl.Tensor[[ROWS, COLS], pl.FP32],
    y: pl.Tensor[[ROWS, COLS], pl.FP32],
    prev_tids: pl.Array[N, pl.TASK_ID],
    out_tids: pl.Array[N, pl.TASK_ID],
) -> pl.Tensor[[ROWS, COLS], pl.FP32]:
    # consumer of the PREVIOUS iteration's carry -> makes `carry` truly loop-carried
    with pl.at(level=pl.Level.CORE_GROUP, name_hint="consumer",
               deps=[prev_tids[i] for i in range(N)]):
        y[0:ROWS, 0:COLS] = pl.add(y[0:ROWS, 0:COLS], 1.0)
    # produce N tids INSIDE manual_scope and yield them into the carried array
    with pl.manual_scope():                       # <-- remove this wrapper -> compiles fine
        for n in pl.parallel(N):
            with pl.at(level=pl.Level.CORE_GROUP, name_hint="prod",
                       deps=[prev_tids[n]]) as prod_tid:
                c0 = n * TILE
                y[0:ROWS, c0:c0 + TILE] = pl.add(y[0:ROWS, c0:c0 + TILE], 1.0)
            out_tids[n] = prod_tid                # <-- scalar yield to array carry
    return y

@pl.jit
def repro(x: pl.Tensor[[ROWS, COLS], pl.FP32], y: pl.Out[pl.Tensor[[ROWS, COLS], pl.FP32]]):
    carry = pl.array.create(N, pl.TASK_ID)
    with pl.at(level=pl.Level.CORE_GROUP, name_hint="seed") as seed_tid:
        y[0:ROWS, 0:COLS] = pl.add(x[0:ROWS, 0:COLS], 0.0)
    for s in pl.unroll(N):
        carry[s] = seed_tid
    for _i in pl.range(LOOP):                     # LOOP > 1 forces cross-iteration carry
        y = layer(x, y, carry, carry)
    return y

if __name__ == "__main__":
    set_backend_type(BackendType.Ascend910B)
    repro.compile_for_test(torch.randn([ROWS, COLS]), torch.empty([ROWS, COLS]))
```

**Contrast (compiles clean):** delete the `with pl.manual_scope():` line and de-indent the `for n in pl.parallel(N):` block by one level (producer in the auto-dep region). The kernel is otherwise identical and compiles without error — isolating `pl.manual_scope()` as the trigger.

### Expected Behavior

Either:
1. codegen supports loop-carrying a `manual_scope`-produced `TaskId` into an `Array[TASK_ID]` (register such tids in `manual_task_id_map_` / honor C++ lexical scope lifetime, in line with the #1577 fix discussion), or
2. the frontend rejects this pattern with a clear, user-facing error — not an `INTERNAL_CHECK` (`Internal error: ...`).

### Actual Behavior

Orchestration codegen aborts:

```
Failed to generate orchestration 'repro': Internal error: scalar yield to array carry
must resolve to a TaskId variable registered in manual_task_id_map_
Check failed: tid_binding != nullptr at src/codegen/orchestration/orchestration_codegen.cpp:1389
```

(Via `compile_for_test` the error is printed but not raised; via the normal device/codegen path it surfaces as a fatal `PartialCodegenError`.)

### Git Commit ID

3ec5e838 (pypto, main)

### NPU Kind

N/A (not hardware-specific)

### Host Platform

Linux (aarch64)

### Additional Context

Related: #1577 (closed — complementary "read" direction of the same `manual_task_id_map_` registration gap).

| Component | Version |
|---|---|
| pypto | `3ec5e838` (branch: `main`) |
| pypto runtime (submodule) | `2668be5a` |
| pypto-lib | `556f615` |
| pto-isa | `b9122ec` |
| ptoas | `v0.45` |
| CANN | not detected |

### PR DESCRIPTION
**Pull request by lyfne123**
## Summary

Fixes #1811.

Orchestration codegen restored `array_carry_vars_` wholesale on every `PTO2_SCOPE(MANUAL)` exit. A `pl.parallel` array carry registered **inside** a `manual_scope` reuses the **enclosing** `pl.range` loop's backing array, but the wholesale restore wiped that registration. The enclosing loop's `YieldStmt` — emitted *after* the manual-scope block — then could not resolve the carry and tripped an `INTERNAL_CHECK`:

```
Internal error: scalar yield to array carry must resolve to a TaskId variable
registered in manual_task_id_map_
```

The identical kernel without the `with pl.manual_scope():` wrapper compiles cleanly — the loop-body AUTO scope restores *after* the yield, not before it.

## Fix

The wholesale restore is correct only for scope-local C++ identifiers (a `PTO2TaskId` local declared inside the block dies at the closing brace — issue #1577). An array carry whose backing array was declared in the enclosing scope survives the brace and is legitimately referenced after it.

On `MANUAL`-scope exit, codegen now preserves array carries whose backing storage is **enclosing-scope-valid** (named by an identifier not in the scope's local set) and reverts only the scope-local ones. The matching per-slot dep names in `manual_task_id_map_` are kept in sync. The pass remains O(N) over the carry map.

## Testing

- [x] New regression test `test_manual_scope_tid_loop_carried_into_array_carry` (confirmed it crashes without the fix and emits correct single-backing-array threading with it)
- [x] `tests/ut/codegen/` — 466 passed
- [x] Full UT suite — 6181 passed, 2 skipped
- [x] Code review completed
- [x] Documentation updated (EN + zh-CN orchestration codegen docs)

## Related Issues

Fixes #1811. Complementary to #1577 (the read-path direction of the same `manual_task_id_map_` registration gap).

### CODE REVIEW

File: tests/ut/codegen/test_orchestration_codegen.py [hunk 1]
```diff
@@ -5660,6 +5660,98 @@ def main(
...
+        carry_arr = decls[0]
+        # The manual-scope parallel loop writes each producer TaskId back into
+        # its slot of the SAME backing array (the per-slot loop-carry write).
+        assert "PTO2_SCOPE(PTO2ScopeMode::MANUAL)" in code, code
+        assert re.search(rf"{carry_arr}\[\w+\]\s*=\s*\w+;", code), code
+        # The enclosing consumer reads every carry slot to fence on the
+        # previous iteration's producers — proof the carry is loop-carried
+        # (and that the per-slot writes feed the next iteration's deps).
+        slot_reads = re.findall(rf"=\s*{carry_arr}\[\w+\];", code)
+        assert len(slot_reads) >= N, code
```
[the remark this answers is not in the document]
Comment (lyfne123) [reply tests/ut/codegen/test_orchestration_codegen.py]: Hardened in `a530b69f`. Note the suggested `len(decls) == 1` would false-fail here: the consumer's `params_t1_deps[N]` / `_submit_deps_buf[N]` are also `PTO2TaskId[N]`, so the carry can't be identified by array size. Instead the carry is now pinned by its seed broadcast (`<arr>[k] = seed_tid;`, unique to the loop carry), its single backing declaration is asserted, and the per-slot write check is scoped to the `PTO2_SCOPE(MANUAL)` region with a variable slot index so it can't be satisfied by the pre-loop seed writes.

**Review by Hzfengsy — APPROVED** [review]
